File: MusicShop.Model/Requests/AmplifierInsertRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicShop.Model.Requests
{
    public class AmplifierInsertRequest
    {
        public string? ProductNumber { get; set; }
        public int? BrandId { get; set; }
        public string? Model { get; set; }
        public decimal? Price { get; set; }
        public string? Description { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public int? Voltage { get; set; }
        public int? PowerRating { get; set; }
        public bool? HeadphoneJack { get; set; }
        public bool? Usbjack { get; set; }
        public int? NumberOfPresets { get; set; }
    }
}

File: UI/musicshop_mobile/lib/models/amplifier/amplifier_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'amplifier_insert_request.g.dart';

@JsonSerializable()
class AmplifierInsertRequest {
  int? brandId;
  String? model;
  double? price;
  String? description;
  int? voltage;
  int? powerRating;
  bool? headphoneJack;
  bool? usbjack;
  int? numberOfPresets;
  String? productImage;

  AmplifierInsertRequest();

  factory AmplifierInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$AmplifierInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$AmplifierInsertRequestToJson(this);
}
